src/Lecture3.hs

```diff
@@ -213,8 +228,18 @@ types that can have such an instance.
 -- instance Foldable Weekday where
 -- instance Foldable Gold where
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+
+instance Foldable List1 where
+    foldr f z (List1 x []) = f x z
+    foldr f z (List1 x xs) = f x (foldr f z xs)
```

Comment:
You don't need to handle case with an empty list separately because `foldr` already returns `z` for empty lists 🆗